src/plans/details/tabs/Mappings/hooks/usePlanMappingsHandlers.ts

```diff
@@ -0,0 +1,162 @@
+import { useState } from 'react';
+import type { InventoryNetwork } from 'src/modules/Providers/hooks/useNetworks';
+import type { InventoryStorage } from 'src/modules/Providers/hooks/useStorages';
+import { StorageClassAnnotation } from 'src/plans/create/steps/storage-map/constants';
+
+import type {
+  OpenShiftNetworkAttachmentDefinition,
+  OpenShiftStorageClass,
+  V1beta1NetworkMap,
+  V1beta1StorageMap,
+} from '@kubev2v/types';
+import { CONDITION_STATUS } from '@utils/constants';
+
+import {
+  mapSourceNetworksIdsToLabels,
+  mapSourceStoragesIdsToLabels,
+} from '../utils/mapMappingsIdsToLabels';
+import {
+  createOnAddNetworkMapping,
+  createOnAddStorageMapping,
+  createOnDeleteMapping,
+  createOnReplaceMapping,
+  createReplacedNetworkMap,
+  createReplacedStorageMap,
+} from '../utils/planMappingsHandlers';
+import type { Mapping } from '../utils/types';
+
+import { usePlanMappingsState, type UsePlanMappingsStateResult } from './usePlanMappingState';
+
+type PlanMappingsHandlers = {
+  canAddNetwork: boolean;
+  canAddStorage: boolean;
+  onAddNetwork: () => void;
+  onAddStorage: () => void;
+  onDeleteNetwork: (mapping: Mapping) => void;
+  onDeleteStorage: (mapping: Mapping) => void;
+  onReplaceNetwork: ({ current, next }: { current: Mapping; next: Mapping }) => void;
+  onReplaceStorage: ({ current, next }: { current: Mapping; next: Mapping }) => void;
+} & UsePlanMappingsStateResult;
+
+type PlanMappingsHandlersParams = {
+  planNetworkMaps: V1beta1NetworkMap;
+  planStorageMaps: V1beta1StorageMap;
+  sourceNetworks: InventoryNetwork[];
+  sourceStorages: InventoryStorage[];
+  targetNetworks: OpenShiftNetworkAttachmentDefinition[];
+  targetStorages: OpenShiftStorageClass[];
+};
+
+type UsePlanMappingsHandlers = (params: PlanMappingsHandlersParams) => PlanMappingsHandlers;
+
+export const usePlanMappingsHandlers: UsePlanMappingsHandlers = ({
+  planNetworkMaps,
+  planStorageMaps,
+  sourceNetworks,
+  sourceStorages,
+  targetNetworks,
+  targetStorages,
+}) => {
+  const mappingsState = usePlanMappingsState(planNetworkMaps, planStorageMaps);
```

Comment:
what is this state? shouldnt this be handled by the form?

Reply:
since splitted the state and the handlers, I wanted to utilize the state in the handlers and pass it on to the form, maybe rename it to mappingPageState or something similar?